pynxtools/dataconverter/convert.py

```diff
@@ -86,30 +93,58 @@ def convert(input_file: Tuple[str],
         nxdl_path = os.path.join(definitions_path, "contributed_definitions", f"{nxdl}.nxdl.xml")
         if not os.path.exists(nxdl_path):
             nxdl_path = os.path.join(definitions_path, "applications", f"{nxdl}.nxdl.xml")
+        if not os.path.exists(nxdl_path):
+            nxdl_path = os.path.join(definitions_path, "base_classes", f"{nxdl}.nxdl.xml")
         if not os.path.exists(nxdl_path):
             raise FileNotFoundError(f"The nxdl file, {nxdl}, was not found.")
 
-    nxdl_root = ET.parse(nxdl_path).getroot()
+    return ET.parse(nxdl_path).getroot(), nxdl_path
 
-    if undocumented:
-        logger.setLevel(UNDOCUMENTED)
+
+def transfer_data_into_template(input_file,
+                                reader, nxdl_name,
+                                nxdl_root: Optional[ET.Element] = None,
+                                **kwargs):
+    """Transfer parse and merged data from input experimental file, config file and eln.
+
+    Experimental and eln files will be parsed and finally will be merged into template.
+    Before returning the template validate the template data.
+
+    Parameters
+    ----------
+    input_file : Union[tuple[str], str]
+        Tuple of files or file
+    reader: str
+        Name of reader such as xps
+    nxdl_root : ET.element
+        Root element of nxdl file, otherwise provide nxdl_name
+    nxdl_name : str
+        Root name of nxdl file, e.g. NXmpes from NXmpes.nxdl.xml
```

Comment:
```suggestion
    nxdl_name : str
        Root name of nxdl file, e.g. NXmpes from NXmpes.nxdl.xml
    nxdl_root : ET.element
        Root element of nxdl file, otherwise provide nxdl_name
```